sorc/chgres_cube.fd/input_data.F90

```diff
@@ -2749,6 +2763,28 @@ subroutine read_input_atm_grib2_file(localpet)
        print*,'tracer ',vlev, maxval(dummy2d),minval(dummy2d)
        dummy3d(:,:,vlev) = real(dummy2d,esmf_kind_r8)
      enddo
+! Jili Dong interpolation for missing levels 
+     if (is_missing .gt. 0 .and. trim(method) .eq. 'intrp') then
+         print *,'intrp tracer '//trim(vname)
+       do jj = 1, j_input
+         do ii = 1, i_input
+             dummy3d_col_in=dummy3d(ii,jj,:)
+             call dint2p(rlevs,dummy3d_col_in,lev_input,rlevs,dummy3d_col_out,    &
+                        lev_input, 2, intrp_missing, intrp_ier) 
+             if (intrp_ier .gt. 0) then
```

Comment:
chgres_cube has an 'official' error handling routine. Call it here (routine error_handler).

Reply:
And below.

Reply:
error_handler added